### ISSUE TITLE
lr_scheduler for language modeling task (cosine vs inverse_sqrt)

### ISSUE THREAD

**Issue by nicolabertoldi** [pytorch/fairseq#852]
I see that in the example "wikitext-103" for the language modeling task
the lr_scheduler is set to **cosine**, while usually for translation task is set to **inverse_sqrt**.

Is there any reason for this?
Is the **inverse_sqrt** a reasonale scheduler?

Any comment is very welcome.

**Comment by myleott** [comment]
Inverse square root was used in the Attention is All You Need paper. Cosine was introduced in this paper: https://arxiv.org/abs/1608.03983. Both are reasonable, but they have different strategies for decreasing the learning rate. In practice a well-tuned cosine schedule typically performs better than inverse square root, but the downside it that you have to define the number of steps in advance.

Inverse square root is a good default that produces decent results.
